src/qamqpclient.cpp

```diff
@@ -660,14 +660,20 @@ QAmqpExchange *QAmqpClient::createExchange(int channelNumber)
 QAmqpExchange *QAmqpClient::createExchange(const QString &name, int channelNumber)
 {
     Q_D(QAmqpClient);
+    if (!name.isEmpty() && d->exchanges.contains(name)
+                    && (!d->exchanges[name].isNull()))
```

Comment:
No you're not meant to subclass it, I was rather pointing to it as an example of a helper utility that <does something> with QObject destroyed signals. So basically its just a QObject that tracks those signals (so we're not cluttering our code with slots for handling that stuff, when we really don't want to think about it).

So there's a few ways to go about it: first (more easily) you can just create your own sort of like QAmqpChannelCleanupHandler/QAmqpChannelCache that you add channels to, and it makes sure to remove them from the internal list when they are destroyed. Another more elegant solution might be to create a QAmqpChannelCache like so:

```
class QAmqpChannelCache : public QObject, public QList<QAmqpChannel*>
{
    Q_OBJECT
public:
    .....

private Q_SLOTS:
    void handleChannelDestruction() {
        QAmqpChannel *channel = qobject_cast<QAmqpChannel*>(sender());
        if (contains(channel)) {   // we ARE a QList, so we can check ourselves here
            remove(channel);
        }
    }
};
```